enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,303 @@
+- Allow workers services access to services like kubedns and apiserver from hosted control plane.
+
+To accomplish the goals listed above the ovn-kubernetes default network has to implement some features triggered for an Hypershift KubeVirt VM.
+hypershift kubevirt VM is detected.
+
+## Motivation
+
+To reach GA at hypershift kubevirt provider it has to implement live migration and adding live-migration features to ovn-k default pod network
+is possible without much impact. This will allow users to spin hypershift kubevirt hosted clusters out of the box without setting up 
+the network at a secondary interface.
+
+### User Stories
+
+I, a Hypershift user, need to migrate one hosted cluster without affecting different ones and have minimal
+disruption.
+
+### Goals
+
+Hypershift kubevirt provider needs the live-migration functionality, missing features at CNI need to be implement at the ovn-kuberentes 
+which is the default CNI for openshift, this should work with dual stack.
+
+### Non-Goals
+
+n/a
+
+## Proposal
+
+At kubevirt every VM is executed inside a "virt-launcher" pod; how the "virt-launcher" pod net interface is "bound" to the VM is specified by the network binding at the VM spec.
+
+The main VM network binding mechanism are:
+- masquerade binding: the VM has a stable fake address that is masqueraded (SNAT) at virt-launcher pod. 
+- bridge binding: The virt-launcher interface is replaced by a bridge with one leg on the pod interface and the other at the VM tap device. When IPAM is configured on the pod interface, a DHCP server is started to advertise the pod's IP to DHCP aware VMs.
+
+Since Masquerade binding has a fake address at VM guest it cannot be used for hypershift, the kubernetes nodes should see the address they are using to register to the api server.
+
+The bridge binding is able to expose the pod ip to the VM as expected by hypershift but kubevirt does not support live migration with it, for kubevirt to support live migration with bridge binding 
+at primary interface the OVN kubernetes CNI need to do the following:
+- not set the IP at pod netns
+- not expect the IP to be on the pod netns
+
+
+### Workflow Description
+
+The gist of the design is:
+
+1. Hypershift kubevirt workers attach to ovn-k node switches but CNI does not set IP addresses in virt-launcher pods netns
+2. DHCP is configured at VMs logical switch ports to deliver the address to VMs
+3. A point to point routing is used so one node's subnet ip can be routed from different node
+3. The VM's gateway ip and mac are independent of the node they are running on using proxy arp
+4. Current OVN arp proxy feature is missing following features: configure MAC and IPv6, they are being implemented [here](https://github.com/ovn-org/ovn/pull/176)
+
+
+### API Extensions
+
+n/a
+
+## Design Details
+
+### OVN-Kubernetes 
+
+#### Topology
+
+**Point to point routing:**
+
+Do point to point routing with a policy for outbound traffic and a static route for
```

Comment:
so the policy is in place so that egress traffic can leave on the original node where the pod was assigned? Why is this necessary? To not disrupt egress/egress reply traffic?

Reply:
Added the following to the explanation 

```
also outbound traffic will will go    
over the node where VM is running on to reduce inter node communication.
```